From 3d207e1c137b7f41256b8bd80622ae95c25335b4 Mon Sep 17 00:00:00 2001
From: Alexander Pivovarov <apivovarov@gmail.com>
Date: Wed, 13 May 2015 17:42:12 -0700
Subject: [PATCH 0304/1164] CDH-31950: HIVE-10665 Continue to make
 udaf_percentile_approx_23.q test more stable
 (Alexander Pivovarov, reviewed by Jason Dere and
 Swarnim Kulkarni)

---
 .../clientpositive/udaf_percentile_approx_23.q     |    2 +-
 .../clientpositive/udaf_percentile_approx_23.q.out |    6 +++---
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/ql/src/test/queries/clientpositive/udaf_percentile_approx_23.q b/ql/src/test/queries/clientpositive/udaf_percentile_approx_23.q
index b345b37..028acdb 100644
--- a/ql/src/test/queries/clientpositive/udaf_percentile_approx_23.q
+++ b/ql/src/test/queries/clientpositive/udaf_percentile_approx_23.q
@@ -90,7 +90,7 @@ set hive.cbo.enable=false;
 -- NaN
 explain
 select percentile_approx(case when key < 100 then cast('NaN' as double) else key end, 0.5) from bucket;
-select percentile_approx(case when key < 100 then cast('NaN' as double) else key end, 0.5) from bucket;
+select percentile_approx(case when key < 100 then cast('NaN' as double) else key end, 0.5) between 340.5 and 343.0 from bucket;
 
 -- with CBO
 explain
diff --git a/ql/src/test/results/clientpositive/udaf_percentile_approx_23.q.out b/ql/src/test/results/clientpositive/udaf_percentile_approx_23.q.out
index 022a884..14a6874 100644
--- a/ql/src/test/results/clientpositive/udaf_percentile_approx_23.q.out
+++ b/ql/src/test/results/clientpositive/udaf_percentile_approx_23.q.out
@@ -556,15 +556,15 @@ STAGE PLANS:
       Processor Tree:
         ListSink
 
-PREHOOK: query: select percentile_approx(case when key < 100 then cast('NaN' as double) else key end, 0.5) from bucket
+PREHOOK: query: select percentile_approx(case when key < 100 then cast('NaN' as double) else key end, 0.5) between 340.5 and 343.0 from bucket
 PREHOOK: type: QUERY
 PREHOOK: Input: default@bucket
 #### A masked pattern was here ####
-POSTHOOK: query: select percentile_approx(case when key < 100 then cast('NaN' as double) else key end, 0.5) from bucket
+POSTHOOK: query: select percentile_approx(case when key < 100 then cast('NaN' as double) else key end, 0.5) between 340.5 and 343.0 from bucket
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@bucket
 #### A masked pattern was here ####
-341.5
+true
 PREHOOK: query: -- with CBO
 explain
 select percentile_approx(key, 0.5) from bucket
-- 
1.7.9.5

